Props Convert

ssorallen opened the pull request:
- `attrForSelected` becomes `attr-for-selected` for uniform dash to
  camel case conversion
- `toggleAttribute` becomes `drawerToggleAttribute` so it doesn't
  override the function by the same name that is added to all Polymer
  elements
- Start computed funcs with `compute` per style guide

Commit:
Convert to uniform `properties` object

* `attrForSelected` becomes `attr-for-selected` for uniform dash to
  camel case conversion
* `toggleAttribute` becomes `drawerToggleAttribute` so it doesn't
  override the function by the same name that is added to all Polymer
  elements

Commit:
Start computed funcs with `compute` per style guide

cdata commented on core-drawer-panel.html:
```diff
@@ -188,157 +149,212 @@
-        responsiveWidth: String,
+        disableEdgeSwipe: {
+          type: Boolean,
+          value: false
```
If boolean properties are always false by default, is it necessary or useful to specify a default? Just wondering..

ssorallen replied:
Are they false by default? I'm not sure. The verbosity seems worthwhile here because the initial state doesn't seem obvious to me with the current API.

cdata replied:
I believe they are always falsey as a rule (an absent attribute implies false). I agree about obviousness. However, if that's the convention we should probably also give `disableSwipe` a default value.

ssorallen replied:
Good points.

cdata commented on core-drawer-panel.html:
```diff
@@ -188,157 +149,212 @@
 
       ready: function() {
+        // Whether the browser has support for the transform CSS property.
+        this.hasTransform = ('transform' in this.style);
```
Why isn't this in `properties`?

ssorallen replied:
Because :penguin:.

cdata commented on core-drawer-panel.html:
```diff
@@ -188,157 +149,212 @@
+        this.hasTransform = ('transform' in this.style);
+
+        // Whether the browser has support for the will-change CSS property.
+        this.hasWillChange = ('willChange' in this.style);
```
Ditto

ssorallen replied:
:bear:

cdata commented on core-drawer-panel.html:
```diff
@@ -188,157 +149,212 @@
 
-          narrow: false,
+        style += 'left:' + ((narrow || rightDrawer) ? '0' : drawerWidth) + ';';
+        style += 'right:' + (rightDrawer ? (narrow ? '' : drawerWidth) : '') + ';';
```
Nested ternaries are not my favorite thing in the world to read. Can we break some of this stuff apart?

Commit:
Let boolean props default to `false`

Omitting an attribute means "false" in HTML. Providing a default has no
effect, so don't add it.

Commit:
Move all possible props to `properties` object

Calculate these with value functions so `this` is available.

Commit:
Split double ternary for easier reading and braining

Nested ternary operators are tough for brains to read.

Commit:
Add a button to flip the drawer in demo for testing

Flip the drawer from right to left by clicking "flip drawer". This helps
to test the right vs. left drawer functionality.

Commit:
Add missing closing div, sloppy HTML

cdata commented:
Should `demo.html` go in `demo/index.html`?

ssorallen commented:
Aha, it looks like demo should go there. I'll move it.

Commit:
Move demo to demo/index.html per repo org guide

* Adjust relative path to point to the component in the parent directory
  so the demo still works.

Commit:
Update copyrights to current year, 2015
